server/service/src/sync/translations/requisition.rs

```diff
@@ -607,4 +652,74 @@ mod tests {
             assert_eq!(translated[0].record.record_data["name_ID"], json!("name_a"));
         }
     }
+
+    #[actix_rt::test]
+    async fn test_sanitise() {
+        let translator = RequisitionTranslation {};
+
+        let (_, connection, _, _) = setup_all("test_sanitise", MockDataInserts::none()).await;
+
+        let sync_record = SyncBufferRow {
+            data: r#"
+            {
+                "//": "Status is set to sent, should be changed to draft",
+                "om_status": "SENT",
+
+                "ID": "50AB29075A7A4A1DA3CDB34465244A61",
+                "authorisationStatus": "none",
+                "colour": 0,
+                "comment": "From request requisition 1 (Approved by test. Email: - and Phone Number: -.)",
+                "custom_data": null,
+                "date_entered": "2024-07-01",
+                "date_order_received": "2024-07-01",
+                "date_required": "0000-00-00",
+                "date_stock_take": "2024-07-01",
+                "daysToSupply": 30,
+                "donor_ID": "",
+                "isRemoteOrder": false,
+                "is_emergency": false,
+                "lastModifiedAt": 1719800437,
+                "linked_purchase_order_ID": "",
+                "linked_requisition_id": "01906c17-0f49-7c13-b6ab-6fab0abb1d20",
+                "name_ID": "947274E3A24D4900996CA516379A1FFD",
+                "nsh_custInv_ID": "",
+                "orderType": "",
+                "periodID": "",
+                "previous_csh_id": "",
+                "programID": "",
+                "requester_reference": "From request requisition 1",
+                "requisition_category_ID": "",
+                "serial_number": 0,
+                "status": "sg",
+                "store_ID": "8659A64D2CF245A1B1BCC7C8F7CDC577",
+                "thresholdMOS": 0,
+                "type": "response",
+                "user_ID": ""
+            }
+            "#.to_string(),
+            ..Default::default()
+        };
+
+        let mut op = assert_variant!(
+            translator.try_translate_from_upsert_sync_record(&connection, &sync_record),
+            Ok( PullTranslateResult::IntegrationOperations(out)) => out
+        );
+
+        let mut upsert =
+            assert_variant!(op.pop(), Some(IntegrationOperation::Upsert(out, _)) => out);
+
+        let requisition_row = upsert
+            .as_mut_any()
+            .and_then(|any| any.downcast_mut::<RequisitionRow>())
```

Comment:
there is an example of downcasting in repository/lib under the trait (with a test)